### PR TITLE
chore(docs): add note about const strings

### PR DESCRIPTION
**Pull request by neuronull**

### CODE REVIEW

File: docs/DEVELOPING.md [hunk 1]
```diff
@@ -229,6 +231,19 @@ a panic here is acceptable.
...
+When re-typing raw string literals repeatedly, this can lead to typo errors,
+especially when names ares simiilar. In general, when reasonable, it is preferred
+to use [Compile-time constants](https://doc.rust-lang.org/std/keyword.const.html)
+when dealing with non-dynamic strings. For example, when working with field names
+for event metadata.
+
+This has not always been a consistently enforced code style for the project, so
+despite seeing usages of raw strings, please take the opportunity when writing
+new code or modifying existing, to use constants or update existing raw strings
+to use constants.
```
Comment (jszwedko) [inline docs/DEVELOPING.md:245]: Could we clarify this to say "strings used more than once?"

Comment (jszwedko) [reply docs/DEVELOPING.md:245]: I also realized that maybe this makes more sense in the style guide? https://github.com/vectordotdev/vector/blob/master/STYLE.md

Comment (bruceg) [inline docs/DEVELOPING.md:245]:
```suggestion
As this has not always been a consistently enforced code style for the project,
please take the opportunity, when modifying existing code, to update existing
raw strings to use constants.
```

**Review by jszwedko — APPROVED** [review]
🎉

**Pull request merged**